src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1468,9 +1474,21 @@ private void TrackNullableStateForAssignment(BoundNode node, Symbol assignmentTa
 
                 if (targetType.IsReferenceType)
                 {
+                    var local = assignmentTarget as LocalSymbol;
+                    if (inferNullability)
+                    {
+                        _variableIsNullable[local] = !valueIsNotNull;
```

Comment:
>_variableIsNullable[local] = !valueIsNotNull; [](start = 24, length = 45)

Do we want to execute this code when local is null?

Reply:
Added assert.